src/com/company/Main.java

```diff
@@ -166,22 +166,38 @@ public static void inOutCheckCounter() {
                 //here we simply fetch the first string which is the status of book
                 String status = record[0];
 
-                if(status.equals("checkOut"))
+                if(status.equalsIgnoreCase("checkOut"))
                 {
                     /*if book status is checkout then we reduce the available count by 1 and increase the
                      checkout counter by 1 */
 
                     book = library.findBook(Integer.parseInt(record[1]));
-                    book.availableCount--;
-                    book.checkoutCounter++;
+                    try {
+                        if (book != null) {
+                            book.availableCount--;
+                            book.checkoutCounter++;
+                        }
+                    }catch (NullPointerException e){
+                        System.out.println("Null pointer exception caught for variable book");
+                    }
```

Comment:
no need for try catch. just check for null before accessing availableCount & checkoutCounter 
if not null : 
    then access
else 
    print invalid book id found in prompt file.